spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/StringBasedJdbcQuery.java

```diff
@@ -189,7 +190,7 @@ private void convertAndAddParameter(MapSqlParameterSource parameters, Parameter
 		Assert.notNull(type, "@Query parameter type could not be resolved");
 
 		JdbcValue jdbcValue;
-		if (value instanceof Iterable) {
+		if (value instanceof Collection && resolvableType.hasGenerics()) {
```

Comment:
It would make sense to follow our general conventions to include `Iterable` and `Streamable` types but exclude custom domain types that implement `Iterable`.

See https://github.com/spring-projects/spring-data-commons/blob/main/src/main/java/org/springframework/data/util/TypeDiscoverer.java#L118-L126 for further reference. Also, we should consider `CustomCollections` to support usage of e.g. vavr collection types as follow up.